testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/realm/RealmTest.java

```diff
@@ -997,4 +1001,43 @@ public void testNoUserProfileProviderComponentUponRealmChange() {
         // this used to return non-empty collection
         assertThat(adminClient.realm(realmName).components().query(null, UserProfileProvider.class.getName()), empty());
     }
+
+    @Test
+    public void testSetEmptyAttributeValues() {
+        String realmName = "testSetEmptyAttributeValues";
+        RealmRepresentation rep = new RealmRepresentation();
+        rep.setRealm(realmName);
+        rep.setAttributes(new HashMap<>());
+        rep.getAttributes().put("myboolean", "");
+        rep.getAttributes().put("mylong", "");
+        rep.getAttributes().put("myint", "");
+        rep.getAttributes().put(RealmAttributes.ACTION_TOKEN_GENERATED_BY_USER_LIFESPAN + ".something", "");
+
+        adminClient.realms().create(rep);
+
+        try {
+            adminClient.realm(realmName);
+
+            testingClient.server().run(session -> {
+                RealmModel realm = session.realms().getRealmByName(realmName);
+                Assert.assertTrue(realm instanceof org.keycloak.models.cache.infinispan.RealmAdapter);
+
+                Assert.assertNull(realm.getUserActionTokenLifespans().get("something"));
+                Assert.assertEquals(true, realm.getAttribute("myboolean", true));
+                Assert.assertEquals(Long.valueOf(123), realm.getAttribute("mylong", (long) 123));
+                Assert.assertEquals(Integer.valueOf(1234), realm.getAttribute("myint", 1234));
+
+                RealmProvider delegate = session.getProvider(CacheRealmProvider.class).getRealmDelegate();
+                RealmModel realm2 = delegate.getRealm(realm.getId());
+                Assert.assertTrue(realm2 instanceof org.keycloak.models.jpa.RealmAdapter);
+
+                Assert.assertNull(realm2.getUserActionTokenLifespans().get("something"));
+                Assert.assertEquals(true, realm2.getAttribute("myboolean", true));
+                Assert.assertEquals(Long.valueOf(123), realm2.getAttribute("mylong", (long) 123));
+                Assert.assertEquals(Integer.valueOf(1234), realm2.getAttribute("myint", 1234));
+            });
+        } finally {
+            adminClient.realm(realmName).remove();
+        }
+    }
```

Comment:
@stianst Should we use something more fancy for main?
```suggestion
    public void testSetEmptyAttributeValues() {
        final String realmName = "testSetEmptyAttributeValues";
        try (Creator<RealmResource> c = Creator.create(adminClient, RealmBuilder.create()
                        .name(realmName)
                        .attribute("myboolean", "")
                        .attribute("mylong", "")
                        .attribute("myint", "")
                        .attribute(RealmAttributes.ACTION_TOKEN_GENERATED_BY_USER_LIFESPAN + ".something", "")
                        .build()
        )) {
            testingClient.server().run(session -> {
                RealmModel realm = session.realms().getRealmByName(realmName);
                Assert.assertTrue(realm instanceof org.keycloak.models.cache.infinispan.RealmAdapter);

                Assert.assertNull(realm.getUserActionTokenLifespans().get("something"));
                Assert.assertEquals(true, realm.getAttribute("myboolean", true));
                Assert.assertEquals(Long.valueOf(123), realm.getAttribute("mylong", (long) 123));
                Assert.assertEquals(Integer.valueOf(1234), realm.getAttribute("myint", 1234));

                RealmProvider delegate = session.getProvider(CacheRealmProvider.class).getRealmDelegate();
                RealmModel realm2 = delegate.getRealm(realm.getId());
                Assert.assertTrue(realm2 instanceof org.keycloak.models.jpa.RealmAdapter);

                Assert.assertNull(realm2.getUserActionTokenLifespans().get("something"));
                Assert.assertEquals(true, realm2.getAttribute("myboolean", true));
                Assert.assertEquals(Long.valueOf(123), realm2.getAttribute("mylong", (long) 123));
                Assert.assertEquals(Integer.valueOf(1234), realm2.getAttribute("myint", 1234));
            });
        }
    }
```

Reply:
I'd rather just merge it as is, and if you want to polish the test afterwards feel free ;)